### PR TITLE
Improved D8 + Composer - CI instructions

### PR DESCRIPTION
**Pull request by alexfornuto**
Closes: DOC-278

## Summary

**[Integrated Composer](https://pantheon.io/docs/integrated-composer)** - Added a section on setting up a Drupal 8 site with Integrated Composer.

**[Drupal 8 and Composer on Pantheon Without Continuous Integration](https://pantheon.io/docs/guides/drupal-8-composer-no-ci)** - Removed this guide and redirected to the new section noted above.

## Remaining Work
(Remove if not needed)
The following changes still need to be completed:

- [ ] List any outstanding work here

## Post Launch

- [ ] Redirect `/docs/guides/drupal-8-composer-no-ci` => `/docs/integrated-composer#drupal-8-with-integrated-composer` **IMPORTANT**
- [ ] Update Addsearch indexing.
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6197](https://pr-6197--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.00000000000001` | `56.99999999999999` | `1` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `88` | `88` | `0` |

Lighthouse Audit Reports:

- [`pr-6197` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/25544/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/25544/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6197--pantheon-docs.my.pantheonfrontend.website/docs)

**Comment by dficker** [comment]
I'm looking at https://pr-6197--pantheon-docs.my.pantheonfrontend.website/docs/integrated-composer as I think that's the preview of the page.

I noticed that when I created a site with the Drupal 9 upstream, it actually put it in SFTP mode right away. I had to switch it to Git mode.

Also, when I try to run the Composer command, it doesn't actually work:

> $ composer require drupal/core-composer-scaffold:^8.8 drupal/core-recommended:^8.8 pantheon-systems/drupal-integrations:^8
> ./composer.json has been updated
> Loading composer repositories with package information
> Updating dependencies (including require-dev)
> Your requirements could not be resolved to an installable set of packages.
> 
>   Problem 1
>     - pantheon-upstreams/upstream-configuration dev-master requires pantheon-systems/drupal-integrations ^9 -> satisfiable by pantheon-systems/drupal-integrations[9.0.0, 9.0.0-alpha1, 9.0.0-alpha2] but these conflict with your requirements or minimum-stability.
>     - pantheon-upstreams/upstream-configuration dev-master requires pantheon-systems/drupal-integrations ^9 -> satisfiable by pantheon-systems/drupal-integrations[9.0.0, 9.0.0-alpha1, 9.0.0-alpha2] but these conflict with your requirements or minimum-stability.
>     - Installation request for pantheon-upstreams/upstream-configuration * -> satisfiable by pantheon-upstreams/upstream-configuration[dev-master].
> 
> 
> Installation failed, reverting ./composer.json to its original content.

Maybe I'm missing a step or something. Or not using the right version of Composer, maybe? But it seems to be insinuating that the "pantheon-upstreams/upstream-configuration" repo is blocking me from switching to D8 because it requires D9?

**Comment by alexfornuto** [comment]
Thanks for testing @dficker! I've pinged @populist to look at the error you got.

**Comment by populist** [comment]
I was able to reproduce your error @dficker and believe the solution is to run the composer require command *inside* the upstream-configuration directory.

File: source/content/integrated-composer.md
Comment (greg-1-anderson) [inline source/content/integrated-composer.md:31]: We don't need to tell folks to back up to the older version of core-composer-scaffold; they're better off staying on ^9 for this one.

File: source/content/integrated-composer.md
Comment (greg-1-anderson) [inline source/content/integrated-composer.md:35]: Shouldn't need to run `composer install` before committing and pushing.

**Comment by populist** [comment]
Looking to add a sentence about how its an easy D9 update and explaining we aren't supporting a D8 IC upstream.

**Comment by dficker** [comment]
I tried following these instructions on a test site, and the install screen still says 9.1.3:

Site: https://dev-ficker-d8-ic-test3.pantheonsite.io/core/install.php
Dashboard: https://admin.dashboard.pantheon.io/sites/3b4d4727-8e5c-46c3-ac34-3382ec49415f

So not sure what I'm doing wrong.

**Comment by greg-1-anderson** [comment]
Did your `composer require` run without error? Did it produce modifications to the composer.lock file? If you run `composer info drupal/core*` do you see Drupal 8 for core and core-recommended? Did you commit and push composer.lock?

**Comment by dficker** [comment]
Yeah, I `cd upstream-configuration` then ran this:
`composer require drupal/core-composer-scaffold:^8.8 \
drupal/core-recommended:^8.8 pantheon-systems/drupal-integrations:^8`

In that folder, I can see core Drupal 8 installed, I guess:

> $ composer info drupal/core*
> drupal/core                   8.9.13 Drupal is an open source content management platform powering millions of websites and applica...
> drupal/core-composer-scaffold 8.9.13 A flexible Composer project scaffold builder.
> drupal/core-recommended       8.9.13 Locked core dependencies; require this project INSTEAD OF drupal/core.

So I committed and push all the changes in the `upstream-configuration` folder, but that doesn't seem to affect the main install or something? The only changes were in the `upstream-configuration` folder but wouldn't something need to be changed in the main `composer.json` or `composer.lock` file as well?

**Comment by greg-1-anderson** [comment]
Ah, yes, obviously I did not step through these instructions myself. The instructions *do* say "From a terminal working in the "upstream-configuration" directory", but I didn't notice that. It would be better to write up these instructions to work from the project root, because it's more steps to modify the upstream directory. Also, an individual site is *not supposed* to modify this directory, whose contents are reserved for use by the upstream.

The issue is, though, that we can't constrain the Drupal version from the top-level composer.json file, because this PR has not been deployed yet:

https://github.com/pantheon-systems/drupal-project/pull/6

@mbaynton would it be possible to get a new release of drupal-project soon?

**Comment by mbaynton** [comment]
@ari-gold currently indicates when the changes should be released

**Comment by populist** [comment]
Thanks for getting the D8 permission change merged @greg-1-anderson . This made the documentation a lot simpler. Just run _composer require drupal/core-recommended:^8.8 pantheon-systems/drupal-integrations:^8_ from the root directory. Updated the PR.

File: source/content/integrated-composer.md
Comment (populist) [reply source/content/integrated-composer.md]: Makes sense to simplify. Pantheon will do this work for you.

**Review by populist — APPROVED** [review]
Updated to be simpler. Looking good to me now.

**Comment by greg-1-anderson** [comment]
n.b. Actual sites will take their upstreams from https://github.com/pantheon-upstreams/drupal-project/blob/master/upstream-configuration/composer.json, which does not have the change yet.

I'd still keep the documentation in its current form, recommending changing the root composer.json on the assumption that there will be a deploy of the Drupal 9 upstream soon.

cc @ari-gold

**Comment by greg-1-anderson** [comment]
@alexfornuto The current documented method does not work with the most recent published version of the upstream. Documenting how to do it with the published version would be awkward, so we should just block until the published upstream is fixed to match the documentation. That will happen in Core Services' current sprint (within the next two weeks).

**Comment by greg-1-anderson** [comment]
Just updating this PR per slack discussion: the core team had updated the upstream, so work may continue here. We are still blocked on deploying until the Mariadb update is generally available.

**Comment by dficker** [comment]
OK, yes, those instructions work. But a few notes:

After these instructions are done, only your local checkout is the Drupal 8 code. It seems to me at least this next step is missing:

Step 4. Run `git push` to deploy the changes to Pantheon.

The Drupal 9 instructions walk you through installing Drupal and getting a local checkout with the Composer libraries up and running. If this is to do the same things, it seems like it should also have these steps at the end in additon:

Step 5. In the  Dev tab on the Dashboard, set the site's Development Mode to SFTP.

Step 6. Still in the Dev tab, click Visit Development Site and follow the prompts to complete the CMS installation.

Step 7. Return to the Dev tab, set the site's Development Mode to Git.

Then you're really ready to keep developing with Integrated Composer, in my opinion.

**Comment by populist** [comment]
Perhaps this should be condensed with https://github.com/pantheon-systems/documentation/pull/6145? It is basically the same process, but a little more fleshed out.

**Comment by greg-1-anderson** [comment]
@EdwardAngert Is this PR obsolete no that #6389 has been merged?

**Comment by EdwardAngert** [comment]
since we're not directing people towards D8, I'm closing this one (and setting a redirect for the D8 No CI doc it referenced, which is already gone but currently 404ing)

**Pull request closed without merging**
